core/domain/exp_jobs_one_off_test.py

```diff
@@ -239,6 +240,45 @@ def _run_batch_job_once_and_verify_output(
                         getattr(expected_job_output[exp_id], prop))
 
 
+class OneOffExplorationFirstPublishedJobTest(test_utils.GenericTestBase):
+
+    EXP_ID = 'exp_id'
+
+    def setUp(self):
+        super(OneOffExplorationFirstPublishedJobTest, self).setUp()
+
+    def test_first_published_time_of_exploration_that_is_unpublished(self):
+        """This tests that the set first published datetime job for exploration
+        takes the first published datetime,regardless if the exploration is
+        published, unpublished, and then republished multiple times.
+        """
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.login(self.ADMIN_EMAIL)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.set_admins([self.ADMIN_EMAIL])
+
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.login(self.OWNER_EMAIL)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+        exploration = self.save_new_valid_exploration(
+            self.EXP_ID, self.owner_id, end_state_name='End')
+        rights_manager.publish_exploration(self.owner_id, self.EXP_ID)
```

Comment:
Would be nice to add a job run at this point, and check that the last published time coincides with the value set by the job. Also this means that you'll have an explicit value to compare the later run to.